packages/components/addon/components/hds/table/index.js

```diff
@@ -179,4 +182,58 @@ export default class HdsTableIndexComponent extends Component {
       onSort(this.sortBy, this.sortOrder);
     }
   }
+
+  @action
+  onSelectionAllChange(checkbox) {
+    const checkboxes = this.allTbodyCheckboxes();
+    this.changeAllRowsSelection(checkboxes, checkbox.checked);
+
+    let { onSelectionChange } = this.args;
+    if (typeof onSelectionChange === 'function') {
+      onSelectionChange(checkboxes);
+    }
+  }
+
+  @action
+  onSelectionRowChange(checkbox) {
+    this.setTableState();
+    let { onSelectionChange } = this.args;
+    if (typeof onSelectionChange === 'function') {
+      onSelectionChange(checkbox);
+    }
+  }
+
+  allTbodyCheckboxes() {
+    const tableBody = document.querySelector(`#${this.tableId} tbody`);
+    const tableBodyRows = Array.from(tableBody.rows);
+    let checkboxes = [];
+
+    tableBodyRows.forEach((row) => {
+      const checkbox = row.querySelector('.hds-table__checkbox');
+      checkboxes.push(checkbox);
+    });
+    return checkboxes;
+  }
+
+  changeAllRowsSelection(checkboxes, checked) {
+    checkboxes.forEach((checkbox) => {
+      checkbox.checked = checked;
+    });
+  }
+
+  setTableState() {
+    const tableHeader = document.querySelector(`#${this.tableId} thead`);
+    const tableHeaderCheckbox = tableHeader.querySelector(
+      '.hds-table__checkbox'
+    );
```

Comment:
these two selectors can be merged

```suggestion
    const tableHeaderCheckbox = document.querySelector(
      #${this.tableId} thead .hds-table__checkbox`
    );
```